plugins/repoclosure.py

```diff
@@ -86,9 +99,12 @@ def _get_unresolved(self, arch=None):
         if self.opts.check:
             checkpkgs = set()
             for repo in self.opts.check:
-                for pkgs_filtered in available.filter(reponame=repo):
+                for pkgs_filtered in self.base.sack.query().available().filter(reponame=repo):
                     checkpkgs.add(pkgs_filtered)
-            pkgs.intersection_update(checkpkgs)
+            if self.opts.pkglist:
+                pkgs.intersection_update(checkpkgs)
```

Comment:
Really? I thought intersection_update would reduce the pkgs set (which already only contains the requested --pkg) to only the ones from the --check repos. Am I wrong? I'm by no means a fluent python programmer.